File: 백준/Bronze/2979. 트럭 주차/트럭 주차.js
const input = require("fs").readFileSync("/dev/stdin").toString().trim().split("\n")

const price = input[0].split(' ').map(Number)
const times = input.slice(1).map(row=>row.split(' ').map(Number))
const visited = Array.from({length:101},()=>0)

for(let [s,e] of times){
    for(let i=s;i<e;i++){
        visited[i] +=1
    }
}
let answer =0

for(let visit of visited){

    if(visit != 0){
        answer += price[visit-1] * visit
    }
}
console.log(answer)

File: 백준/Bronze/2979. 트럭 주차/트럭 주차.py
import sys

input = sys.stdin.read().strip().split("\n")
price = list(map(int, input[0].split()))
times = [list(map(int, row.split())) for row in input[1:]]

visited = [0] * 101

for s, e in times:
    for i in range(s, e):
        visited[i] += 1

answer = 0
for visit in visited:
    if visit != 0:
        answer += price[visit - 1] * visit

print(answer)
